src/modules/networking/src/lib/Networking.cpp

```diff
@@ -391,53 +402,84 @@ void NetworkingObjectBase::GenerateInterfaceTypesMap()
             }
         }
 
-        interfaceData.clear();
-        interfaceName.clear();
-        interfaceType.clear();
         interfaceTypesData = nextDataToProcess;
     }
 
-    if (!usingNetworkManager)
+    if ((this->m_interfaceTypesMap.empty()) && (this->m_networkManagementService == NetworkManagementService::NetworkManager))
+    {
+        this->m_networkManagementService = NetworkManagementService::Unknown;
+    }
+}
+
+void NetworkingObjectBase::GetInterfaceTypesFromSystemdNetworkd()
+{
+    std::string interfaceTypesData = RunCommand(g_getInterfaceTypesNetworkctl);
+    std::stringstream interfaceTypesDataStream(interfaceTypesData);
+    std::string line;
+    while(std::getline(interfaceTypesDataStream, line))
     {
-        interfaceTypesData = RunCommand(g_getInterfaceTypesNetworkctl);
-        std::stringstream interfaceTypesDataStream(interfaceTypesData);
-        while(std::getline(interfaceTypesDataStream, interfaceData))
+        if (this->m_networkManagementService == NetworkManagementService::Unknown)
+        {
+            this->m_networkManagementService = NetworkManagementService::SystemdNetworkd;
+        }
+
+        std::regex interfaceTypesDataPatternNetworkctl("^\\s*[0-9]+\\s+.*$");

```

Comment:
>"^\\s*[0-9]+\\s+.*$" [](http://example.com/codeflow?start=55&length=20)

Make this a constant (same as any other such incline constant strings or characters or numbers)